provider/vsphere/internal/vsphereclient/createvm_test.go

```diff
@@ -384,6 +384,36 @@ func (s *clientSuite) TestCreateVirtualMachineRootDiskSize(c *gc.C) {
 	})
 }
 
+func (s *clientSuite) TestVerifyMAC(c *gc.C) {
+	var testData = []struct {
+		Mac    string
+		Result bool
+	}{
+		{"foo:bar:baz", false},
+		{"00:22:55:11:34:11", false},
+		{"00:50:56:123:11:11", false},
+		{"00:50:56:40:12:23", false},
+		{"00:50:56:3f:ff:ff", true},
+		{"00:50:56:12:34:56", true},
+		{"00:50:56:2A:eB:Cd", true},
+	}
+	for i, test := range testData {
+		c.Logf("test #%d: MAC=%s expected %s", i, test.Mac, test.Result)
+		c.Check(VerifyMAC(test.Mac), gc.Equals, test.Result)
+	}
+}
+
+// GenerateMAC is random so we verify a few times if the MACs returned are
+// are correct.
+func (s *clientSuite) TestGenerateMAC(c *gc.C) {
+	for i := 0; i < 100; i++ {
+		mac, err := GenerateMAC()
+		c.Assert(err, jc.ErrorIsNil)
+		c.Logf("Test #%d: %s", i, mac)
+		c.Check(VerifyMAC(mac), gc.Equals, true)
```

Comment:
would it be better to do:

c.Check(VerifyMAC(mac), jc.IsTrue, gc.Commentf("GenerateMAC() created an invalid MAC address: %v", mac))
Then you don't end up with 100 lines of "success" when you're just looking for the failure.